Source/Core/Core/BootManager.cpp

```diff
@@ -144,6 +145,17 @@ bool BootCore(std::unique_ptr<BootParameters> boot, const WindowSystemInfo& wsi)
 
   Core::System::GetInstance().Initialize();
 
+  // DirectoryBlobs (including Riivolution-patched discs) may end up larger than a real physical Wii
+  // disc, which triggers Error #001. In those cases we manually make the check succeed to avoid
+  // problems.
+  const bool should_fake_error_001 =
+      StartUp.bWii &&
+      (!boot->riivolution_patches.empty() ||
+       (std::holds_alternative<BootParameters::Disc>(boot->parameters) &&
+        std::get<BootParameters::Disc>(boot->parameters).volume->GetBlobReader().GetBlobType() ==
+            DiscIO::BlobType::DIRECTORY));
+  Config::SetCurrent(Config::SESSION_SHOULD_FAKE_ERROR_001, should_fake_error_001);
```

Comment:
Ideally this should be implemented in such a way that if the user switches discs, the hack turns on or off to match the new disc.

Reply:
Oh right, we do support mid-game disc swapping...